### DISCUSSION TITLE
Monitor as a Singleton?

### DISCUSSION THREAD

**Discussion by ruffk** [braddwalker/ZwiftPacketMonitor#20] [Q&A] [answered]
Brad

I want to have a single instance of the Monitor class feeding events to n number of other classes running in there own threads.  I'm considering wrapping the Monitor in a Singleton class.  So,

1) Do you think this will work?   
2) Is it even advisable to have more than one instance of Monitor running anyway?  If not, maybe it should be added as Singleton instead of Transient.

TIA

Kevin

**Comment by braddwalker** [comment, marked as answer]
@ruffk that's a great catch -- it really should be a singleton and not transient. I really can't think of a usecase for needing multiple instances of the Monitor class itself. You should be able to instantiate it as a singleton, inject it into any classes/threads, and have the various consumers all just hook into the event listeners.

**Reply by braddwalker** [reply]
Have added issue #21 to track this.

**Reply by ruffk** [reply]
One more thought regarding this.  Since it will be a singleton, it would be nice to have an IsStarted property to check.  You might want to use that internally too to make sure someone doesn't call StartCaptureAsync while it's already running (to at least ignore the request).
